src/coreclr/jit/fgbasic.cpp

```diff
@@ -4840,11 +4840,34 @@ BasicBlock* Compiler::fgSplitBlockAfterNode(BasicBlock* curr, GenTree* node)
             }
         }
 
-        curr->bbCodeOffsEnd = max(curr->bbCodeOffs, splitPointILOffset);
+        if (splitPointILOffset == BAD_IL_OFFSET)
+        {
+            // Try to look forwards in the next block
+            for (GenTree* node : LIR::AsRange(newBlock))
```

Comment:
Does LIR::AsRange always guarantee in HIR that it will walk trees in the execution order?

Reply:
This function is only used for LIR (there is an assert at the top)